astropy/io/ascii/cparser.pyx

```diff
@@ -738,6 +741,9 @@ cdef class CParser:
     def set_names(self, names):
         self.names = names
 
+    def get_header_names(self):
+        return self.header_names
+
```

Comment:
This seems to be a feature of cython where (given the current declarations) `self.header_names` is not directly accessible outside the class.  I'm not an expert with Cython but I think I tried direct access and it failed.